pkg/vpc/vpc.go

```diff
@@ -65,6 +74,56 @@ func SetSubnets(vpc *api.ClusterVPC, availabilityZones []string) error {
 	return nil
 }
 
+func SplitInto16(parent *net.IPNet) ([]*net.IPNet, error) {
+	networkLength, _ := parent.Mask.Size()
+	networkLength += 4
+
+	var subnets []*net.IPNet
+	for i := 0; i < 16; i++ {
+		ip4 := parent.IP.To4()
+		if ip4 != nil {
+			n := binary.BigEndian.Uint32(ip4)
+			n += uint32(i) << uint(32-networkLength)
+			subnetIP := make(net.IP, len(ip4))
+			binary.BigEndian.PutUint32(subnetIP, n)
+
+			subnets = append(subnets, &net.IPNet{
+				IP:   subnetIP,
+				Mask: net.CIDRMask(networkLength, 32),
+			})
+		} else {
+			return nil, fmt.Errorf("Unexpected IP address type: %s", parent)
+		}
+	}
+
+	return subnets, nil
+}
+
+func SplitInto8(parent *net.IPNet) ([]*net.IPNet, error) {
```

Comment:
Sorry forgot to include comment, this could be reduced into `SplitIntoN`

Reply:
@michaelbeaumont I'm a bit hesitant to do that, while I managed to manipulate the first function to accurately produce splits of 16 instead of 8, I don't know how to ensure the same change for any given input, e.g. splitting into 10.

Reply:
Yes, it should work for any power of two (and only for powers of two).